json-endpoints/class.wpcom-json-api-comment-endpoint.php

```diff
@@ -127,7 +127,7 @@ function get_comment( $comment_id, $context ) {
 				);
 				break;
 			case 'author' :
-				$response[$key] = (object) $this->get_author( $comment, 'edit' === $context && current_user_can( 'edit_comment', $comment->comment_ID ) );
+				$response[$key] = (object) $this->get_author( $comment, in_array( $context, array( 'edit', 'display' ) ) && current_user_can( 'edit_comment', $comment->comment_ID ) );
```

Comment:
I was weary to drop the `context` check altogether for compatibility, and because I think that we'll have to check the usage of the class and evaluate if we need a `context` at all (my guess is that we don't). But the `switch` argument is very compelling, so, I've dropped the context check on 7de64c10ccff185a27e209085c59ac4418c0dc83.

Ready for review!